### PR TITLE
apps: Add missing AppStream metadata paths

### PR DESCRIPTION
**Pull request by martinpitt**
Thanks to Fabian Vogt @Vogtinator for pointing out!

See https://github.com/cockpit-project/cockpit/issues/19902#issuecomment-1987334718

### CODE REVIEW

File: pkg/apps/watch-appstream.py [hunk 1]
```diff
@@ -356,8 +356,12 @@ def installed_callback(path):
     def available_callback(path):
         process_file(path, lambda path, xml: db.notice_available(path, xml))
 
-    watcher.watch_directory('/usr/share/metainfo', installed_callback)
+    # https://www.freedesktop.org/software/appstream/docs/chap-CatalogData.html
     watcher.watch_directory('/usr/share/swcatalog/xml', available_callback)
+    watcher.watch_directory('/var/cache/swcatalog/xmls', available_callback)
+    watcher.watch_directory('/var/lib/swcatalog/xml', available_callback)
+    # legacy paths
+    watcher.watch_directory('/usr/share/metainfo', installed_callback)
```
Comment (Vogtinator) [inline pkg/apps/watch-appstream.py:364]: `/usr/share/metainfo` is not legacy, it's still the current location for metainfo of installed packages

File: pkg/apps/watch-appstream.py [hunk 2]
```diff
@@ -356,8 +356,12 @@ def installed_callback(path):
     def available_callback(path):
         process_file(path, lambda path, xml: db.notice_available(path, xml))
 
-    watcher.watch_directory('/usr/share/metainfo', installed_callback)
+    # https://www.freedesktop.org/software/appstream/docs/chap-CatalogData.html
     watcher.watch_directory('/usr/share/swcatalog/xml', available_callback)
+    watcher.watch_directory('/var/cache/swcatalog/xmls', available_callback)
```
Comment (Vogtinator) [inline pkg/apps/watch-appstream.py:361]:
```suggestion
    watcher.watch_directory('/var/cache/swcatalog/xml', available_callback)
```

Comment (martinpitt) [reply pkg/apps/watch-appstream.py]: Whoops, thanks!

File: pkg/apps/watch-appstream.py (hunk 1 above)
Comment (martinpitt) [reply pkg/apps/watch-appstream.py]: Thanks, moved to the right place.

**Review by Vogtinator — APPROVED** [review]

**Review by jelly — APPROVED** [review]
Thanks!

**Pull request merged**